stonybook/pipeline/quote/quote.py

```diff
@@ -0,0 +1,961 @@
+                        prefixTokens = self.tokensBeforeInSameSentence(quote.start_token.tokenId - 1,
+                                                                       quote.start_token.sentenceId)
+                        quoteTokens = prefixTokens
+
+            # Next Token
+            quoteTokens += [self.tokenDict[x] for x in range(quote.quote_start, quote.quote_end + 1)]
+            if quote.end_token.tokenId + 1 in self.tokenDict:
+                nextToken = self.tokenDict[quote.end_token.tokenId + 1]
+                prevToken = self.tokenDict[quote.end_token.tokenId - 1]
+                if nextToken.paragraphId == quote.end_token.paragraphId:
+                    if prevToken.text != "." and nextToken.sentenceId == quote.end_token.sentenceId:
+                        quoteTokens += self.tokensAfterInSameSentence(quote.end_token.tokenId + 1,
+                                                                      quote.end_token.sentenceId)
+                    elif prevToken.text != "." and nextToken.sentenceId > quote.end_token.sentenceId:
+                        quoteTokens += self.tokensAfterInSameSentence(quote.end_token.tokenId + 1,
+                                                                      quote.end_token.sentenceId + 1)
+
+            candidates = []
+
+            def outsideQuote(token):
+                if token.tokenId < quote.quote_start or token.tokenId > quote.quote_end:
+                    return True
+                return False
+
+            for token in quoteTokens:
+                if outsideQuote(token):
+                    if (token.pos.startswith("VB") and token.lemma.lower() in self.reportingVerbs):
+                        candidates.append(token)
+
+            def getTokenDistance(token):
+                if token.tokenId < quote.quote_start:
+                    return abs(token.tokenId - quote.quote_start)
+                return abs(token.tokenId - quote.quote_end)
+
+            mention = None
+
+            if len(candidates) == 0:
+                self.quotes[i].mention = None
+                continue
+
+            candidates.sort(key=lambda x: getTokenDistance(x))
+            reportingVerb = candidates[0]
+
+            def expand_noun(tok):
+                compounds = [
+                    child
+                    for tc in [tok]
+                    for child in tc.children
+                    if child.dep == "compound"
+                ]
+                return [tok] + compounds
+
+            children = [child for child in reportingVerb.children]
+            filtered_children = [x for x in children if
+                                 x.tokenId < quote.start_token.tokenId or x.tokenId > quote.end_token.tokenId]
+            childrenPOS = [child.pos for child in filtered_children]
+            mention_start = mention_end = -1
+            if 'NNP' in childrenPOS:
```

Comment:
Instead of having if-else blocks with the same code, just write a for-loop that iterates over the POS tags you are interested in